sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnectionFactory.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs
+{
+
+    /// <summary>
+    ///   A factory to create a connection to the Azure Event Hubs service, enabling client communications with a specific
+    ///   Event Hub instance within an Event Hubs namespace.  A single connection may be shared among multiple
+    ///   Event Hub producers and/or consumers, or may be used as a dedicated connection for a single
+    ///   producer or consumer client.
+    /// </summary>
+    public class EventHubConnectionFactory
```

Comment:
The inspiration for this class came from attempting to mock out the connection behavior. This could be done without a dedicated factory type, I just thought it would be cleaner with a dedicated factory.